Stage1/environment/env_1d.py

```diff
@@ -61,30 +91,48 @@ def reset(self):
         self.state = [self.drone_coordinates[0], self.drone_coordinates[1], np.random.rand() * 2 * np.pi, 0]
         return np.array(self.state)
 
-    def render(self, mode='human'):
+    def render(self, mode='human', plan_background=None):
         screen_width = self.plan.shape[0]
         screen_height = self.plan.shape[1]
 
         if self.viewer is None:
             from gym.envs.classic_control import rendering
             self.viewer = rendering.Viewer(screen_width, screen_height)
-
-        # TODO: add plan
-        # gym doesn't work with images, only primitive forms, so need to find a way to draw a np.array
-        # np.array can be generated by show_layers
+            plan = NPImage(self.plan_file_path, self.plan, screen_width, screen_height)
+            self.viewer.add_geom(plan)
+            drone = rendering.FilledPolygon([(-20, -5), (-20, 5), (0, 0)])
+            self.dronetrans = rendering.Transform()
+            drone.add_attr(self.dronetrans)
+            self.viewer.add_geom(drone)
+
```

Comment:
Please provide comments to the method body in order to simplify code understanding.

Reply:
I will muddy the code by comments as OpenAI Gym mechanics understanding are required. I don't want to describe all iceberg where only tip of the iceberg is shown. 
I will explain how OpenAI Gym works for people in slack.